# **Ejercicio 13:  Cultura** #

##### Una persona tiene un libro de cada uno de los autores siguientes: Mario Vargas Llosa, Alberto Ruy Sánchez, Gabriel García Márquez, Günther Grass, Patrick Süskind, Heinrich Böll, Arturo Pérez-Reverte y Camilo José Cela.  #####
#####  Como aún no los ha leído, quiere ponerlos en una repisa cerca de su cama. De acuerdo con esto, determina:  #####
**a)** ¿De cuántas formas puede acomodar los libros sobre la repisa?  
**b)** ¿De cuántas maneras puede ordenarlos si quiere que los libros de autores latinoamericanos estén juntos?  
**c)** ¿De cuántas formas puede acomodarlos, si los autores latinoamericanos deben estar juntos, así como los autores alemanes y los españoles también?  
**d)** Determina las probabilidades correspondientes a los incisos (b) y (c).  

In [1]:
#### a) ####  ¿De cuántas formas puede acomodar los libros sobre la repisa?  
from math import comb, perm, factorial
formasTotalesAcomodar = factorial(8)
print("a) Hay", formasTotalesAcomodar, "maneras de acomodar los libros sobre la repisa ")

#### b) #### ¿De cuántas maneras puede ordenarlos si quiere que los libros de autores latinoamericanos estén juntos?  
# 3 autores son latinos, como deben de estar juntos los tomamos como un elemento
formasTotalesAcomodarLatinos = factorial(6) * factorial(3)
print("b) Hay", formasTotalesAcomodarLatinos, "maneras de ordenar los libros si los autores latinoamericanos están juntos")

#### c) ####  ¿De cuántas formas puede acomodarlos, si los autores latinoamericanos deben estar juntos, así como los autores alemanes y los españoles también?  

# 3 autores son latinos, 2 son españoles y 3 son alemanes
formasTotalesAcomodarLaEsAl = (factorial(3) * factorial(2) * factorial(3)) * factorial(3)
print("c) Hay", formasTotalesAcomodarLaEsAl, "maneras de ordenar los libros si los autores latinoamericanos, alemanes y españoles deben estar juntos")

#### d) ####  Determina las probabilidades correspondientes a los incisos (b) y (c).  
probabilidadB = formasTotalesAcomodarLatinos / formasTotalesAcomodar
probabilidadC = formasTotalesAcomodarLaEsAl / formasTotalesAcomodar
print("d) Probabilidades")
print("   Probabilidad inciso (b):", probabilidadB, "10.71%")
print("   Probabilidad inciso (c):", probabilidadC, "1.07%")

a) Hay 40320 maneras de acomodar los libros sobre la repisa 
b) Hay 4320 maneras de ordenar los libros si los autores latinoamericanos están juntos
c) Hay 432 maneras de ordenar los libros si los autores latinoamericanos, alemanes y españoles deben estar juntos
d) Probabilidades
   Probabilidad inciso (b): 0.10714285714285714 10.71%
   Probabilidad inciso (c): 0.010714285714285714 1.07%


## EJERCICO 13 ##
 
### Cultura ###
Una persona tiene un libro de cada uno de los autores siguientes: Mario Vargas Llosa, Alberto Ruy Sánchez, Gabriel García Márquez, Günther Grass, Patrick Süskind, Heinrich Böll, Arturo Pérez-Reverte y Camilo José Cela.   
Como aún no los ha leído, quiere ponerlos en una repisa cerca de su cama. De acuerdo con esto, determina:  

a) **¿De cuántas formas puede acomodar los libros sobre la repisa?**

Para determinar el número de formas en que puede acomodar los libros, utilizamos la permutación de 8 libros distintos. La fórmula de permutaciones es:

$$
n! = 8!
$$

Donde \( n \) es el número total de libros.

$$
8! = 40320
$$

Por lo tanto, hay 40,320 maneras de acomodar los libros sobre la repisa.

b) **¿De cuántas maneras puede ordenarlos si quiere que los libros de autores latinoamericanos estén juntos?**

Los autores latinoamericanos son Mario Vargas Llosa, Alberto Ruy Sánchez y Gabriel García Márquez. Consideramos el grupo de libros de autores latinoamericanos como una unidad. Entonces, tenemos 6 unidades para ordenar (el grupo de autores latinoamericanos y los otros 5 libros).

El número de formas en que se pueden acomodar estas 6 unidades es:

$$
6!
$$

Dentro del grupo de autores latinoamericanos, los libros pueden ordenarse de:

$$
3!
$$

Entonces, el número total de maneras de ordenarlos es:

$$
6! \times 3! = 720 \times 6 = 4320
$$

Por lo tanto, hay 4,320 maneras de ordenar los libros si los autores latinoamericanos están juntos.

c) **¿De cuántas formas puede acomodarlos, si los autores latinoamericanos deben estar juntos, así como los autores alemanes y los españoles también?**

Los autores alemanes son Günther Grass, Heinrich Böll y Patrick Süskind. Los autores españoles son Arturo Pérez-Reverte y Camilo José Cela. Consideramos cada grupo de autores como una unidad. Entonces, tenemos 3 unidades para ordenar (los grupos de autores latinoamericanos, alemanes y españoles).

El número de formas en que se pueden acomodar estas 3 unidades es:

$$
3!
$$

Dentro del grupo de autores latinoamericanos, los libros pueden ordenarse de:

$$
3!
$$

Dentro del grupo de autores alemanes, los libros pueden ordenarse de:

$$
3!
$$

Dentro del grupo de autores españoles, los libros pueden ordenarse de:

$$
2!
$$

Entonces, el número total de maneras de ordenarlos es:

$$
3! \times 3! \times 3! \times 2! = 6 \times 6 \times 6 \times 2 = 432
$$

Por lo tanto, hay 432 maneras de ordenar los libros si los autores latinoamericanos, alemanes y españoles deben estar juntos.

d) **Determina las probabilidades correspondientes a los incisos (b) y (c).**

La probabilidad de que los libros de autores latinoamericanos estén juntos es:  
El caclulo de la probabilidad  de un evento especifico se calcula mediante el uso de esta formula: 
$$ P(A) = \frac{\text{Número de resultados favorables}}{\text{Número total de resultados posibles}} $$
$$
P(\text{Latinoamericanos juntos}) = \frac{\text{Número de maneras con los latinoamericanos juntos}}{\text{Número total de maneras}} =
 \frac{4320}{40320} =  \approx 0.107143 = 10.71/\%
$$

La probabilidad de que los autores latinoamericanos, alemanes y españoles estén juntos es:

$$
P(\text{Latinoamericanos, alemanes y españoles juntos}) = \frac{\text{Número de maneras con latinoamericanos, alemanes y españoles juntos}}{\text{Número total de maneras}} = \frac{432}{40320} = \frac{1}{93.333} \approx 0.0107 = 1.07\%
$$
